### PR TITLE
add esp32c6

### PR DESCRIPTION
**Pull request by seanaye**
Adds the options for the esp32 c6 dev board in the cargo generate toml

### CODE REVIEW

**Comment by sredman** [comment]
> Hi @seanaye! Thanks for your contribution! We would also need some logic, as the new targets are not supported in some ESP-IDF versions, see https://github.com/espressif/esp-idf#esp-idf-release-and-soc-compatibility.

Do you have a suggestion for how you're thinking of this? I've never edited a cargo-generate template before, but from my struggles it seems that this isn't an easy thing to do.

**Comment by sredman** [comment]
> > Hi @seanaye! Thanks for your contribution! We would also need some logic, as the new targets are not supported in some ESP-IDF versions, see https://github.com/espressif/esp-idf#esp-idf-release-and-soc-compatibility.
> 
> Do you have a suggestion for how you're thinking of this? I've never edited a cargo-generate template before, but from my struggles it seems that this isn't an easy thing to do.

I have this, but it's not pretty, and will grow "infinitely" as more boards are released which need new SDKs.

```
[conditional.'defaults == false && (mcu == "esp32s6" || mcu == "esp32p4")'.placeholders.espidfver]
type = "string"
prompt = "ESP-IDF version (master = UNSTABLE)"
choices = ["master"] # Supported as of v5.1
default = "master"

[conditional.'defaults == false && (mcu == "esp32" || mcu == "esp32c3" || mcu == "esp32s2" || mcu == "esp32s3")'.placeholders.espidfver]
type = "string"
prompt = "ESP-IDF version (master = UNSTABLE)"
choices = ["v4.4", "v5.0", "master"]
default = "v4.4"
```

**Comment by Vollbrecht** [comment]
Also missing: 
as long as the upstream support for the new c6 target is not integrated, the template should include the target manually as a file and invoke it in the build command.  
riscv32imac-esp-espidf.json
```json
{
    "arch": "riscv32",
    "cpu": "generic-rv32",
    "data-layout": "e-m:e-p:32:32-i64:64-n32-S128",
    "eh-frame-header": false,
    "emit-debug-gdb-scripts": false,
    "env": "newlib",
    "features": "+m,+a,+c",
    "is-builtin": false,
    "linker": "riscv32-esp-elf-gcc",
    "llvm-target": "riscv32",
    "max-atomic-width": 64,
    "os": "espidf",
    "panic-strategy": "abort",
    "relocation-model": "static",
    "target-family": [
      "unix"
    ],
    "target-pointer-width": "32",
    "vendor": "espressif"
  }
  ```
  
  invoked with `--target riscv32imac-esp-espidf.json` or better add to the rustflags in config.toml so the user workflow dont change

**Comment by sredman** [comment]
> Also missing: as long as the upstream support for the new c6 target is not integrated, the template should include the target manually as a file and invoke it in the build command. riscv32imac-esp-espidf.json
> 
> <snip>
> 
> invoked with `--target riscv32imac-esp-espidf.json` or better add to the rustflags in config.toml so the user workflow dont change

Support for risc32-imac for esp-espidf was merged a couple of days ago: https://github.com/rust-lang/rust/pull/111369

I have modified a project generated by this template locally, and I can confirm that the Hello World example works with a C6 board given you're using a nightly rust build. So now, ideally, the template should be made to work, and documentation updated.

**Comment by Vollbrecht** [comment]
> > Also missing: as long as the upstream support for the new c6 target is not integrated, the template should include the target manually as a file and invoke it in the build command. riscv32imac-esp-espidf.json
> > 
> > invoked with `--target riscv32imac-esp-espidf.json` or better add to the rustflags in config.toml so the user workflow dont change
> 
> Support for riscv32imac-esp-espidf was merged a couple of days ago: [rust-lang/rust#111369](https://github.com/rust-lang/rust/pull/111369)
> 
> I have modified a project generated by this template locally, and I can confirm that the Hello World example works with a C6 board given you're using a nightly rust build. So now, ideally, the template should be made to work, and documentation updated.

:star_struck: nice, i missed that merge

File: cargo/pre-script.rhai [hunk 1]
```diff
@@ -23,6 +23,24 @@ let targets = #{
         gcc_target: "riscv32-esp-elf",
         wokwi_board: "board-esp32-c3-devkitm-1",
     },
+    esp32c6: #{
+        arch: "riscv",
+        rust_target: "riscv32imc-esp-espidf",
```
Comment (sredman) [inline cargo/pre-script.rhai:28]: This needs to be `riscv32imac-esp-espidf` (imAc) for esp32c6

Comment (sredman) [reply cargo/pre-script.rhai:28]: See https://github.com/esp-rs/esp-idf-template/pull/124

**Comment by sredman** [comment]
> Hi @seanaye! Thanks for your contribution! We would also need some logic, as the new targets are not supported in some ESP-IDF versions, see https://github.com/espressif/esp-idf#esp-idf-release-and-soc-compatibility.

I have put my proposal in https://github.com/esp-rs/esp-idf-template/pull/125

**Pull request merged**